sys-boot/udk/udk-2018.ebuild

```diff
@@ -0,0 +1,256 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 )
+PYTHON_REQ_USE="sqlite"
+
+inherit flag-o-matic multiprocessing python-single-r1 toolchain-funcs
+
+DESCRIPTION="Tianocore UEFI Development kit"
+HOMEPAGE="https://github.com/tianocore/tianocore.github.io/wiki/EDK-II"
+
+if [[ ${PV} == *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/tianocore/edk2.git"
+	KEYWORDS=""
+else
+	MY_V="UDK$(ver_cut 1)"
+	SRC_URI="https://github.com/tianocore/edk2/archive/v${MY_V}.tar.gz -> ${P}.tar.gz
+	doc? ( https://github.com/tianocore/edk2/releases/download/v${MY_V}/${MY_V}.Documents.zip -> ${P}-docs.zip )"
+	S="${WORKDIR}/edk2-v${MY_V}"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="BSD-2"
+SLOT="0"
+IUSE="doc examples"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="strip"
+
+RDEPEND="${PYTHON_DEPS}"
+DEPEND="${RDEPEND}
+	>=dev-lang/nasm-2.14.02
+	>=sys-power/iasl-20160729
+	doc? ( app-arch/unzip )"
+
+DOCDIR="${WORKDIR}/Documents"
+HTML_DOCS="${DOCDIR}"/*
+
+# Generated libs for EFI can contain WX sections
+QA_EXECSTACK="
+	usr/lib*/libBaseLib.a:*
+	usr/lib*/libBaseIoLibIntrinsic.a:*
+"
+
+pkg_setup() {
+	python_setup 'python2.7'
+
+	local uname_arch=$(uname -m | sed -e 's:i[3456789]86:IA32:')
+	if [[ ${uname_arch} == "x86_64" ]] || [[ ${uname_arch} == "amd64" ]] ; then
+		ARCH=X64
+	else
+		ARCH=${uname_arch}
+	fi
+
+	# Select toolchain within predefined ones
+	if tc-is-gcc; then
+		if ver_test $(gcc-version) -ge 5; then
+			TOOLCHAIN_TAG="GCC5"
+		elif ver_test $(gcc-version) -ge 4.9; then
+			TOOLCHAIN_TAG="GCC49"
+		elif ver_test $(gcc-version) -ge 4.8; then
+			TOOLCHAIN_TAG="GCC48"
+		elif ver_test $(gcc-version) -ge 4.7; then
+			TOOLCHAIN_TAG="GCC47"
+		elif ver_test $(gcc-version) -ge 4.6; then
+			TOOLCHAIN_TAG="GCC46"
+		elif ver_test $(gcc-version) -ge 4.5; then
+			TOOLCHAIN_TAG="GCC45"
+		elif ver_test $(gcc-version) -ge 4.4; then
+			TOOLCHAIN_TAG="GCC44"
+		fi
+	elif tc-is-clang; then
+		if ver_test $(clang-version) -ge 3.8; then
+			TOOLCHAIN_TAG="CLANG38"
+		elif ver_test $(clang-version) -ge 3.5; then
+			TOOLCHAIN_TAG="CLANG35"
+		fi
+	fi
+	if [[ -z ${TOOLCHAIN_TAG} ]]; then
+		TOOLCHAIN_TAG="ELFGCC"
+	fi
+}
+
+src_unpack() {
+	unpack ${A}
+
+	local doc_name
+	local f
+	if use doc; then
+		rm -f "${DOCDIR}/"*.chm || die
+		pushd "${DOCDIR}" >/dev/null || die
+		for f in "${DOCDIR}/"*.zip; do
+			doc_name=$(basename "${f}" .zip)
+			unpack "${f}"
+			mv "${DOCDIR}/html" "${DOCDIR}/${doc_name}" || die
+		done
+		popd >/dev/null || die
+		rm -f "${DOCDIR}/"*.zip || die
+	fi
+}
+
+src_configure() {
+	# Compile of Base Tools is required for further setting up the environment
+	# Base tools does not like parallel make
+	local cflags_save=${CFLAGS}
```

Comment:
Instead of all this save/restore logic, just put the extra flags inline in sed:

    sed -e "s:...:\1 ${CFLAGS} -MD -fshort-wchar ...